src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -231,9 +231,17 @@ uint16_t ColorControlServer::computeTransitionTimeFromStateAndRate(ColorControlS
         min = p->currentValue;
     }
 
-    transitionTime = max - min;
-    transitionTime *= 10;
-    transitionTime /= rate;
+    // If currentValue == finalValue, force 1 step
+    if (max == min)
+    {
+        transitionTime = 1;
+    }
+    else
+    {
+        transitionTime = max - min;
+        transitionTime *= 10;
+        transitionTime /= rate;
```

Comment:
It may still happen that `transitionTime` equals to 0 even if `max != min` (say, `max == min + 1` and `rate > 10`). I think we should either add `transitionTime = std::max(transitionTime, 1)` here regardless of other conditions or make sure `transitionTime == 0` is handled correctly by the further code.